connectors/sources/dropbox_aiohttp.py

```diff
@@ -0,0 +1,661 @@
+                self.access_token = response_data["access_token"]
+                self.token_expiration_time = evaluate_timedelta(
+                    seconds=int(response_data["expires_in"]), time_skew=20
+                )
+                self._logger.debug("Access Token generated successfully")
+
+    def check_errors(self, response):
+        error_response = response.get("error")
+        if error_response == "invalid_grant":
+            raise InvalidRefreshTokenException("Configured Refresh Token is invalid.")
+        elif error_response == "invalid_client: Invalid client_id or client_secret":
+            raise InvalidClientCredentialException(
+                "Configured App Key or App Secret is invalid."
+            )
+        else:
+            raise Exception(
+                f"Error while generating an access token. Error: {error_response}"
+            )
+
+    @cached_property
+    def _get_session(self):
+        self._logger.debug("Generating aiohttp client session")
+        timeout = aiohttp.ClientTimeout(total=None)  # pyright: ignore
+
+        return aiohttp.ClientSession(timeout=timeout, raise_for_status=True)
+
+    async def close(self):
+        self._sleeps.cancel()
+        await self._get_session.close()
+        del self._get_session
+
+    def _get_request_headers(self, file_type, **kwargs):
+        kwargs = kwargs["kwargs"]
+        request_headers = {
+            "Authorization": f"Bearer {self.access_token}",
+        }
+        if file_type == FILE:
+            request_headers["Dropbox-API-Arg"] = f'{{"path": "{kwargs["path"]}"}}'
+        elif file_type == PAPER:
+            request_headers[
+                "Dropbox-API-Arg"
+            ] = f'{{"path": "{kwargs["path"]}", "export_format": "markdown"}}'
+        elif file_type == RECEIVED_FILE:
+            request_headers["Dropbox-API-Arg"] = f'{{"url": "{kwargs["url"]}"}}'
+        else:
+            request_headers["Content-Type"] = "application/json"
+        return request_headers
+
+    def _get_retry_after(self, retry, exception):
+        if retry > self.retry_count:
+            raise exception
+        self._logger.warning(
+            f"Retry count: {retry} out of {self.retry_count}. Exception: {exception}"
+        )
+        retry += 1
+        return retry, RETRY_INTERVAL**retry
+
+    async def _handle_client_errors(self, retry, exception):
+        retry, retry_seconds = self._get_retry_after(retry=retry, exception=exception)
+        match exception.status:
+            case 401:
+                await self._set_access_token()
+            case 409:
+                raise InvalidPathException(
+                    f"Configured Path: {self.path} is invalid or not found."
+                ) from exception
+            case 429:
+                response_headers = exception.headers
+                retry_seconds = int(
+                    response_headers.get("Retry-After", DEFAULT_RETRY_AFTER)
```

Comment:
we had a recent bug where we were fetching the header like this, and the casing changed from `Retry-After` to `retry-after` and things broke. Can you make this fetch case insensitive?

Reply:
Sure, i'll make all header-keys lower and then fetch them accordingly.